.github/workflows/golangci-lint.yml

```diff
@@ -13,7 +13,7 @@ jobs:
     steps:
       - uses: actions/setup-go@v2
         with:
-          go-version: 1.17.8
+          go-version: 1.18.5
```

Comment:
Is there any particular reason we specify the patch version?

Reply:
That's what they did last time: https://github.com/aws/eks-anywhere/pull/1841/files#diff-d9a956120ac84289d2650137f64bd8f0893331de05e44cc41899dd984c9e0d48
That said, I just followed the same process and documented it, I don't know if there is a reason or we were just following what was there.

I would like to think it's to ensure consistency and make sure we always use the same version across tests, but maybe not.

@a-cool-train @vivek-koppuru you guys know?

Reply:
If we don't strive for binary reproducibility I don't think this is necessary; leaving the patch version off will reduce maintenance burden.

Reply:
It was initially added by @danbudris here: https://github.com/aws/eks-anywhere/pull/1732

I am not opposed to excluding the patch version. I think that the changes don't change too much between the patch versions to affect this functionality.

Reply:
Changed